api/data/NMBS/vehicleinformation.php

```diff
@@ -207,23 +225,35 @@ private static function getData($html, $lang, $fast)
                     $nextDayArrival = 1;
                 }
                 $previousHour = (int)substr($departureTime, 0, 2);
+                $dateDatetime = DateTime::createFromFormat('dmy', $date);
 
                 $stops[$j] = new Stop();
                 $stops[$j]->station = $station;
                 $stops[$j]->departureDelay = $departureDelay;
                 $stops[$j]->departureCanceled = $departureCanceled;
-                $stops[$j]->scheduledDepartureTime = tools::transformTime('0' . $nextDay . 'd'.$departureTime.':00', date('Ymd'));
-                $stops[$j]->scheduledArrivalTime = tools::transformTime('0' . $nextDayArrival . 'd'.$arrivalTime.':00', date('Ymd'));
+                $stops[$j]->scheduledDepartureTime = tools::transformTime('0' . $nextDay . 'd'.$departureTime.':00', $dateDatetime->format('Ymd'));
+                $stops[$j]->scheduledArrivalTime = tools::transformTime('0' . $nextDayArrival . 'd'.$arrivalTime.':00', $dateDatetime->format('Ymd'));
                 $stops[$j]->arrivalDelay = $arrivalDelay;
                 $stops[$j]->arrivalCanceled = $arrivalCanceled;
                 $stops[$j]->platform = new Platform();
                 $stops[$j]->platform->name = $platform;
                 $stops[$j]->platform->normal = $normalplatform;
                 //for backward compatibility
-                $stops[$j]->time = tools::transformTime('0' . $nextDay . 'd'.$departureTime.':00', date('Ymd'));
+                $stops[$j]->time = tools::transformTime('0' . $nextDay . 'd'.$departureTime.':00', $dateDatetime->format('Ymd'));
                 $stops[$j]->delay = $departureDelay;
                 $stops[$j]->canceled = $departureCanceled;
 
+                // Add occupancy
+                foreach ($occupancyArr as $stopOccupancy) {
+                    if($station->{'@id'} == $stopOccupancy["from"][0]) {
+                        $URI = self::getURLForOccupancy($stopOccupancy["occupancy"]);
+
+                        $stops[$j]->occupancy->{'@id'} = $URI;
+                        $stops[$j]->occupancy->name = basename($URI);
+                        break;
```

Comment:
Isn't breaking out of a for lop not writing very ugly code? Looks like this should have been a while loop instead?